### ISSUE TITLE
[BUG] Pivot view is unwieldy when there are a lot of variables

### ISSUE THREAD

**Issue by colmanflynn** [CSOIreland/PxStat#288]
I've had a few complaints from business areas about this.

E.G. BHA02 -> View Data

Select all fields and then Pivot on Region & County. 
There are 45 variables in this field, so the first handful are listed along the top and the rest are listed down the left hand side.

Is there anyway to make it more useable?

![BHA02 Pivot](https://user-images.githubusercontent.com/59689411/99283198-934e8200-282c-11eb-9116-1f4cf0adb098.PNG)

**Comment by lorenzobruni** [comment]
Any screen/browser, no matter the device, has a limited size in width, while instead is naturally scrollable vertically.

When pivoting by a large amount of variables (ie. 45), then the system is attempting to render 45 or more columns in the same amount of space.
The bigger the screen, the more columns will fit in.
Any column that does not fit in, then will be render _responsively_ instead.

A couple of considerations:

1. What is the purpose to Pivot by a large number of columns? Please note that the table is interactive and users can still filter and sort by column without pivoting... it's just about user-education.
Viewing the data on a browser is about performing basic and simple data queries that would work on any device, such as mobile, tables and also desktops (by providing also full accessibility compliance) o to allow downloading any dataset size.
Complex analysis with a lot o variables require adequate tools, such as spreadsheet software: in this scenario, the user can download the data and review it comfortably in their spreadsheet software.
FYI, there is already a change in progress to allow downloading Pivoted data: https://github.com/CSOIreland/PxStat/issues/252
This should mitigate the issue.

2. Another change that we were considering is to allow toggling a **full screen** view. Perhaps this should be available to desktop users, where they can open large tables on full screen, disabling the responsive design and simulating the behavior of a spreadsheet software by allowing horizontal scrolling.
By the way, rendering hundreds or thousands of pivoted columns even in this scenario will make it unreadable anyway... these are situations where common sense should apply.
FYI, This change has not yet been logged.

**Is the combination for the above items 1 and 2 a viable solution?**

Thanks

FYI @damienderoistecso

**Comment by lorenzobruni** [comment]
@colmanflynn 
Do you have any feedback about the proposed alternative? If good we can workshop on it... Thanks

**Comment by colmanflynn** [comment]
Thanks Lorenzo,


Sorry for the delay in replying, I wanted to talk through the options with the Digital Comms team before I got back to you.


The internal users were pivoting large numbers of columns to have a layout similar to the tables in StatBank, so they could check the data as they had always done.
The main question raised by them was why the option was available if the results would be unwieldy?

I agree user education would help, and they can download the file and pivot the table in Excel if required. 
Also, the users are not as familiar with the concept of responsive pages, even if they do use them regularly.


After speaking with the rest of Digital Comms, we think the option you mentioned of a Full Screen View, with a Scroll Bar, would probably be the best all round.


Thanks,
Colman Flynn,
Digital Communications
------------------------------------------------
 
Central Statistics Office | Skehard Road, | Cork, | T12 X00E | Ireland. 
Tel +353 21 453 5011   
www.cso.ie


Lorenzo Bruni ---18/11/2020 23:15:18---_______________________________________________________________________ | CAUTION: This email origin

**Comment by lorenzobruni** [comment]
Ok, we will investigate. Thanks.

**Comment by colmanflynn** [comment]
We received a comment from an external user from a national agency regarding a issue using Pivot and Downloading to xlsx:

Hi All,

I am looking to download data from PxStat using the new pivot feature. There seems to be an issue when you ask to pivot and then download the data as an xlsx. The data is stored at text in each excel cell which would need to be converted to a number format in excel to be used properly. I have attached an example.

I have played with it and it seems this doesn’t happen if you leave the data un-pivoted.

Could this bug be fixed?
![image](https://user-images.githubusercontent.com/59689411/101153296-24d73580-361c-11eb-9086-4ca02b23efc7.png)
